llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8298,6 +8306,32 @@ static std::optional<bool> isImpliedCondICmps(const ICmpInst *LHS,
       return LPred == RPred;
   }
 
+  // handle R0 = L0 binop V
+  Value *R0Op1 = nullptr;
+  if (match(L1, m_APInt(LC)) && match(R1, m_APInt(RC)) &&
```

Comment:
Maybe init `LC` and `RC` to nullptr above and use the `APInt` matches above? (Although will need to switch the apint matchers to before the `L0 == R0`.